src/Implementation/Shared/Core/AsyncPump.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+// http://blogs.msdn.com/b/pfxteam/archive/2012/01/20/10259049.aspx
+
+namespace Microsoft.Threading
+{
```

Comment:
I would like to, just for clarity. Definitely keep the link to the blog post. You could even comment a bit more on the link and intent (if you think it would add value).